pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -37,45 +37,60 @@ const (
 	enabled        string = "aro.machinehealthcheck.enabled"
 )
 
-type Reconciler struct {
-	log *logrus.Entry
+type MachineHealthCheckReconciler struct {
+	base.AROController
 
 	dh dynamichelper.Interface
-
-	client client.Client
 }
 
-func NewReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *Reconciler {
-	return &Reconciler{
-		log:    log,
-		dh:     dh,
-		client: client,
+func NewMachineHealthCheckReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *MachineHealthCheckReconciler {
+	return &MachineHealthCheckReconciler{
+		AROController: base.AROController{
+			Log:    log,
+			Client: client,
+			Name:   ControllerName,
+		},
+		dh: dh,
 	}
 }
 
-func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	instance := &arov1alpha1.Cluster{}
-	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
+// Reconcile watches MachineHealthCheck objects, and if any changes,
+// reconciles the associated ARO MachineHealthCheck object
+func (r *MachineHealthCheckReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.GetCluster(ctx)
+
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
 	if !instance.Spec.OperatorFlags.GetSimpleBoolean(enabled) {
-		r.log.Debug("controller is disabled")
+		r.Log.Debug("controller is disabled")
 		return reconcile.Result{}, nil
 	}
 
-	r.log.Debug("running")
+	r.Log.Debug("running")
 	if !instance.Spec.OperatorFlags.GetSimpleBoolean(managed) {
+		r.SetProgressing(ctx, "MHC and it's alerts are disabled because this cluster is not MHC managed.")
```

Comment:
I would spell this out to help make it more clear to the customer:
```suggestion
		r.SetProgressing(ctx, "Managed MachineHealthCheck and associated alerts have been disabled in this cluster - reconciling the MachineHealthCheck object.")
```

Reply:
We should not be setting a Progressing status here (or at all in this controller, in its current state). 

Progressing statuses should indicate to us that a reconciliation is actively in progress. In this controller, all the changes are performed in one go (which is an issue in and of itself that will need to be addressed in follow-up work, but not here). 

If the goal is to indicate to SRE that the MHC is unmanaged, the spec block of the cluster resource that these statuses ultimately end up on contains the `aro.machinehealthcheck.managed` flag within its spec.

Reply:
I'll update it by eod today.